src/views/News/index.js

```diff
@@ -19,530 +23,764 @@ import {
+      updateSnackbar('News Posting Failed to Update');
     } else {
-      this.updateSnackbar('News Posting Updated Successfully');
+      updateSnackbar('News Posting Updated Successfully');
     }
 
     // close the window and reload to update data
     // (necessary since data is currently not pulled from render method)
-    this.setState({ openPostActivity: false });
+    setOpenPostActivity(false);
     window.top.location.reload();
   }
 
-  async handleSubmit() {
-    // create the JSON newsItem object to post
+  async function handleSubmit() {
+    let newImage;
+
+    if (showCropper !== null) {
+      let croppedImage = cropperRef.current.cropper
+        .getCroppedCanvas({ width: CROP_DIM })
+        .toDataURL();
+      newImage = croppedImage.replace(/data:image\/[A-Za-z]{3,4};base64,/, '');
+    }
+
+    //There is no else here because if there is no image, the JSON object SHOULD
+    //contain null, since the API has been designed to look for a null case and handle
+    //it, treating it appropriately to mean there is no image for the post.
+
     let newsItem = {
-      categoryID: this.state.newPostCategory,
-      Subject: this.state.newPostSubject,
-      Body: this.state.newPostBody,
+      categoryID: newPostCategory,
+      Subject: newPostSubject,
+      Body: newPostBody,
+      Image: newImage,
     };
 
     // submit the news item and give feedback
     let result = await newsService.submitStudentNews(newsItem);
     if (result === undefined) {
-      this.updateSnackbar('News Posting Failed to Submit');
+      updateSnackbar('News Posting Failed to Submit');
     } else {
-      this.updateSnackbar('News Posting Submitted Successfully');
+      updateSnackbar('News Posting Submitted Successfully');
     }
 
     // close the window and reload to update data
     // (necessary since data is currently not pulled from render method)
-    this.setState({ openPostActivity: false });
+    setOpenPostActivity(false);
     window.top.location.reload();
   }
 
-  // This should be the only time we pull from the database
-  async loadNews() {
-    this.setState({ loading: true });
-
-    if (this.props.authentication) {
-      const newsCategories = await newsService.getCategories();
-      const personalUnapprovedNews = await newsService.getPersonalUnapprovedFormatted();
-      const unexpiredNews = await newsService.getNotExpiredFormatted();
-      this.setState({
-        loading: false,
-        categories: newsCategories,
-        news: unexpiredNews,
-        personalUnapprovedNews: personalUnapprovedNews,
-        filteredNews: unexpiredNews,
-      });
-    } else {
-      // TODO: test authentication handling and neaten code (ex. below)
-      // alert("Please sign in to access student news");
-    }
-  }
+  function imageOnLoadHelper(reader) {
+    var dataURL = reader.result.toString();
+    var i = new Image();
+    i.onload = async () => {
+      if (i.width < CROP_DIM || i.height < CROP_DIM) {
```

Comment:
I don't think we actually need to enforce minimum dimensions for News. That code was leftover from uploading ID Photos where they had to be a minimum size for printing ID cards, but since news images can be whatever the user wants (as long as they get approved), it doesn't matter what size/resolution they are.